### PR TITLE
[#2822] feat(core): support catalog events for event listener

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
support catalog events for event listener

### Why are the changes needed?

Fix: #2822 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
existing tests

### CODE REVIEW

File: core/src/main/java/com/datastrato/gravitino/catalog/CatalogEventDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      throw e;
+    }
+  }
+
+  @Override
+  public Catalog[] listCatalogsInfo(Namespace namespace) throws NoSuchMetalakeException {
+    try {
+      Catalog[] catalogs = dispatcher.listCatalogsInfo(namespace);
+      eventBus.dispatchEvent(
+          new ListCatalogInfoEvent(PrincipalUtils.getCurrentUserName(), namespace));
```
Comment (FANNG1) [inline core/src/main/java/com/datastrato/gravitino/catalog/CatalogEventDispatcher.java:72]: seems no need to create a sperate ` ListCatalogInfoEvent`, we could reuse `ListCatalogEvent`, @jerryshao @mchades WDYT?

**Comment by jerryshao** [comment]
@FANNG1 please review all the missing javadocs and add them all.

File: core/src/main/java/com/datastrato/gravitino/catalog/CatalogEventDispatcher.java (hunk 1 above)
Comment (jerryshao) [reply core/src/main/java/com/datastrato/gravitino/catalog/CatalogEventDispatcher.java:72]: I'm OK with this solution.

**Comment by FANNG1** [comment]
> @FANNG1 please review all the missing javadocs and add them all.

done

File: core/src/main/java/com/datastrato/gravitino/catalog/CatalogEventDispatcher.java (hunk 1 above)
Comment (FANNG1) [reply core/src/main/java/com/datastrato/gravitino/catalog/CatalogEventDispatcher.java]: removed `ListCatalogInfoEvent`

**Review by jerryshao — APPROVED** [review]

**Pull request merged**